.github/workflows/clippy_bors.yml

```diff
@@ -0,0 +1,161 @@
+name: Test Clippy (bors)
+
+on:
+  push:
+    branches: [auto, try]
+    # Don't run Clippy tests, when only textfiles were modified
+    paths-ignore:
+    - 'COPYRIGHT'
+    - 'LICENSE-*'
+    - '**.md'
+    - '**.txt'
+  pull_request:  # TODO: Remove this
+    # Don't run Clippy tests, when only textfiles were modified
+    paths-ignore:
+    - 'COPYRIGHT'
+    - 'LICENSE-*'
+    - '**.md'
+    - '**.txt'
+
+env:
+  RUST_BACKTRACE: 1
+  CARGO_TARGET_DIR: '${{ github.workspace }}/target'
+  GHA_CI: 1
+
+jobs:
+  changelog:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Check Changelog
+      run: |
+        PR=$(echo "${{ github.ref }}" | grep -o "[0-9]*")
+        output=$(curl -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" -s "https://api.github.com/repos/rust-lang/rust-clippy/pulls/$PR" | \
+          python -c "import sys, json; print(json.load(sys.stdin)['body'])" | \
+          grep "^changelog: " | \
+          sed "s/changelog: //g")
+        if [[ -z "$output" ]]; then
+          echo "ERROR: PR body must contain 'changelog: ...'"
+          exit 1
+        elif [[ "$output" = "none" ]]; then
+          echo "WARNING: changelog is 'none'"
+        fi
+  base:
+    needs: changelog
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        host: [x86_64-unknown-linux-gnu, i686-unknown-linux-gnu, x86_64-apple-darwin, x86_64-pc-windows-msvc]
+        exclude:
+        - os: ubuntu-latest
+          host: x86_64-apple-darwin
+        - os: ubuntu-latest
+          host: x86_64-pc-windows-msvc
+        - os: macos-latest
+          host: x86_64-unknown-linux-gnu
+        - os: macos-latest
+          host: i686-unknown-linux-gnu
+        - os: macos-latest
+          host: x86_64-pc-windows-msvc
+        - os: windows-latest
+          host: x86_64-unknown-linux-gnu
+        - os: windows-latest
+          host: i686-unknown-linux-gnu
+        - os: windows-latest
+          host: x86_64-apple-darwin
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2.0.0
+    - name: Install dependencies (Linux-i686)
+      run: |
+        sudo dpkg --add-architecture i386
+        sudo apt-get update
+        sudo apt-get install gcc-multilib libssl-dev:i386 libgit2-dev:i386
+      if: matrix.host == 'i686-unknown-linux-gnu'
+    - name: rust-toolchain
+      uses: actions-rs/toolchain@v1.0.3
```

Comment:
**Just a note**: Look at https://github.com/rust-lang/rust-clippy/runs/403260719#step:4:20, it maybe better to
removing rustc-docs and clippy component before updating nightly toolchain.
Or should we use a helpful script like these:
* https://github.com/lzutao/junction/blob/master/ci/install-rust.bash
* https://github.com/lzutao/junction/blob/master/ci/install-rustup.bash

Reply:
profile used is already minimal. 

> The minimal profile includes as few components as possible to get a working compiler (rustc, rust-std, and cargo). It's recommended to use this component on Windows systems if you don't use local documentation, and in CI.

No idea why it installs rust-docs. Maybe because I also install rustfmt?

Reply:
Oh boy. I first checkout the repo, which includes a `rust-toolchain` file. Then when running rustup show as the first command of `actions-rs/toolchain` it tries to use the nightly toolchain, as specified in `rust-toolchain`. It doesn't find it obviously and installs it, to run `rustup show`. After that it sets the profile to minimal and installs the `nightly` toolchain, which gets skipped, because it is already installed.